### PR TITLE
api: support set store state.

### PR DESCRIPTION
**Pull request by disksing**
Sometime we want to recover store from wrong state. Sometime we want to cancel removing a store.

### CODE REVIEW

File: pdctl/command/store_command.go [hunk 1]
```diff
@@ -50,6 +51,16 @@ func NewDeleteStoreCommand() *cobra.Command {
 	return d
 }
 
+// NewSetStoreStateCommand returns a state subcommand of storeCmd.
+func NewSetStoreStateCommand() *cobra.Command {
+	d := &cobra.Command{
+		Use:   "state <store_id> [Up|Offline|Tombstone]",
```
Comment (siddontang) [inline pdctl/command/store_command.go]: can we set the tombstone state directly? Seem it is dangerous.

Comment (disksing) [reply pdctl/command/store_command.go]: We already have API to let user to set a store's state to 'Tombstone' (from any state)... See: https://github.com/pingcap/pd/blob/73fa4c4f2c394817fbb511afb06a5933b845167e/server/cluster.go#L486-L490

Comment (siddontang) [reply pdctl/command/store_command.go]: Got it.

Comment (siddontang) [reply pdctl/command/store_command.go]: But can we transfer a tombstone store to up again?

Comment (disksing) [reply pdctl/command/store_command.go]: I think we can. The 'tombstone' is just a flag in pd-server that prevent data been transferred to the store, and make the store ignored by any balancers.

File: pdctl/command/store_command.go [hunk 2]
```diff
@@ -105,6 +116,24 @@ func deleteStoreCommandFunc(cmd *cobra.Command, args []string) {
...
+func setStoreStateCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 2 {
+		fmt.Println("Usage: store state <store_id> [Up|Offline|Tombstone]")
+		return
+	}
+	if _, err := strconv.Atoi(args[0]); err != nil {
+		fmt.Println("store_id should be a number")
+		return
+	}
+	prefix := fmt.Sprintf(storePrefix+"/state?state=%s", args[0], args[1])
```
Comment (nolouch) [inline pdctl/command/store_command.go]: I think this way more like GET

**Comment by siddontang** [comment]
LGTM
PTAL @huachaohuang

**Comment by huachaohuang** [comment]
LGTM

**Comment by cofyc** [comment]
I saw the original commit to add state set command in `pd-ctrl`, why it was removed?

**Comment by disksing** [comment]
@cofyc It's a dangerous feature. We only want experts to use it.

**Pull request merged**